dids/src/main/kotlin/web5/sdk/dids/DidIonManager.kt

```diff
@@ -62,6 +75,42 @@ public fun DidIonManager(builderAction: DidIonConfiguration.() -> Unit): DidIonM
 /** [DidIonManager] is sealed, so we provide an impl so the constructor can be called. */
 private class DidIonManagerImpl(configuration: DidIonConfiguration) : DidIonManager(configuration)
 
+/**
+ * The options when updating an ION did.
+ *
+ * @param didString The full did. e.g. `did:ion:123`
+ * @param updateKeyAlias The alias within the key manager that refers to the last update key.
+ * @param servicesToAdd The services to add in the did document.
+ * @param idsOfServicesToRemove Ids of the services to remove from the did document.
+ * @param publicKeysToAdd PublicKeys to add to the DID document.
+ * @param idsOfPublicKeysToRemove Keys to remove from the DID document.
+ */
+public class UpdateDidIonOptions(
+  public val didString: String,
+  public val updateKeyAlias: String,
+  public val servicesToAdd: Iterable<Service>? = null,
```

Comment:
Curious as to why the adds are more generic than the remove (Iterable vs Set). Is order important for add operations?

(https://github.com/TBD54566975/web5-kt/pull/77#discussion_r1354395800)

Reply:
When I looked at the spec it didn't seem like the order was important (or that each type of patch was required to be present)

Reply:
And in addition to https://github.com/TBD54566975/web5-kt/pull/77/files#r1359995250 was confused at marshalling everything into a List at the end - why not just be a List from the start?